src/FSharpLint.Core/Rules/Conventions/UnneededRecKeyword.fs

```diff
@@ -0,0 +1,44 @@
+﻿module FSharpLint.Rules.UnneededRecKeyword
+
+open System
+open FSharp.Compiler.Syntax
+open FSharpLint.Framework.Ast
+open FSharpLint.Framework.Rules
+open FSharpLint.Framework
+open FSharpLint.Framework.Suggestion
+
+let runner (args: AstNodeRuleParams) =
+    match args.AstNode, args.CheckInfo with
+    | AstNode.ModuleDeclaration (SynModuleDecl.Let (isRecursive, bindings, _)), Some checkInfo when isRecursive ->
+        match bindings with
+        | SynBinding (_, _, _, _, _, _, _, SynPat.LongIdent (longDotId, _, _, _, _, range), _, _, _, _) :: _ ->
+            let symbolUses = checkInfo.GetAllUsesOfAllSymbolsInFile()
+            let funcName = longDotId.Lid.Head.idText
```

Comment:
If you expect long identifier to have exactly 1 element, it's better to explicitly pattern match it (`SynLongIdent([ident], _)` instead of `longDotId` in pattern)

Reply:
@webwarrior-ws would we need a new test to cover this?

Reply:
I don't think you can define a function with dotted identifier. So there should always be 1 element in idents list.

Reply:
not sure what you mean but yeah please Mershad address this (and discuss in telegram with Taras if needed)

Reply:
> @Mersho why did you mark this resolved? I don't see a commit addressing it

we discussed about it and there is no failure test I can create for it.

Reply:
that doesn't excuse anything, where is the commit addressing it? (even if it doesn't add a test)

Reply:
It has been addressed in this commit:

https://github.com/fsprojects/FSharpLint/pull/652/commits/aefd4fbd46ef6e97177db7d72f6bbefa5e9962db

i can amend and explain why long identifier have exactly 1 element.